src/Compilers/Core/Portable/Operations/ICollectionElementInitializerOperation.cs

```diff
@@ -28,11 +27,16 @@ public interface ICollectionElementInitializerOperation : IOperation
         /// <summary>
         /// Arguments passed to add method invocation.
         /// </summary>
-        ImmutableArray<IOperation> Arguments { get; }
+        ImmutableArray<IArgumentOperation> Arguments { get; }
 
         /// <summary>
-        /// Flag indicating if this is a dynamic invocation.
+        /// Flag indicating if this is a dynamic invocation. Will always be false.
         /// </summary>
+        /// <remarks>
+        /// This flag is obsolete, and dynamic collection initializers have been broken out into a separate interface,
+        /// <see cref="IDynamicCollectionElementInitializerOperation"/>.
+        /// </remarks>
+        [Obsolete]
         bool IsDynamic { get; }
```

Comment:
This PR appears to remove one API and Obsolete another. Can you explain why?

Reply:
As mentioned in the email thread, I'm currently following the design that seemed to have the most support. This is still subject to design team approval and compat council approval.